tests/tests_unit/test_config.py

```diff
@@ -13,23 +13,60 @@ def test_global_config_singleton(self):
         ):
             GlobalConfig()
 
-    # TODO: Add test for default_client_config
-    def test_apply_settings(self):
+    @pytest.mark.parametrize(
+        "client_config",
+        [
+            {
+                "project": "test-project",
+                "base_url": "https://test-cluster.cognitedata.com/",
+                "credentials": {"token": "abc"},
+                "client_name": "test-client",
+            },
+            ClientConfig(
+                project="test-project",
+                base_url="https://test-cluster.cognitedata.com/",
+                credentials=Token("abc"),
+                client_name="test-client",
+            ),
+            None,
+        ],
+    )
+    def test_apply_settings(self, monkeypatch, client_config):
+        monkeypatch.delattr(GlobalConfig, "_instance")  # ensure that the singleton is re-instantiated
+        gc = GlobalConfig()
+        assert gc.max_workers == 5
+        assert gc.max_retries == 10
+
         settings = {
-            "max_workers": 5,
-            "max_retries": 3,
+            "max_workers": 6,
+            "max_retries": 11,
+            "default_client_config": client_config,
         }
-        global_config.apply_settings(settings)
-        assert global_config.max_workers == 5
-        assert global_config.max_retries == 3
+        gc.apply_settings(settings)
+        assert gc.max_workers == 6
+        assert gc.max_retries == 11
+
+        if client_config:
+            assert isinstance(gc.default_client_config, ClientConfig)
+            assert isinstance(gc.default_client_config.credentials, Token)
+            assert gc.default_client_config.project == "test-project"
+        else:
+            assert gc.default_client_config is None
 
     def test_load_non_existent_attr(self):
         settings = {
-            "test": 10,
+            "max_workers": 0,  # use a nonsensical value to ensure that it is not applied without assuming other tests kept the default value
+            "invalid_1": 10,
+            "invalid_2": "foo",
         }
-        with pytest.raises(ValueError, match=r"Invalid key in global config: .*"):
+
+        with pytest.raises(ValueError, match=r"Invalid keys provided for global_config, no settings applied:"):
             global_config.apply_settings(settings)
 
+        assert (
+            global_config.max_workers != 0
+        )  # confirm that the valid keys were not applied since we don't want a partial application
```

Comment:
```suggestion
        # confirm that the valid keys were not applied since we don't want a partial application
        assert global_config.max_workers != 0
```